pytket/tests/circuit_test.py

```diff
@@ -713,16 +792,25 @@ def test_phase_polybox_big() -> None:
     phase_polynomial = {
         (True, False, True): 0.333,
         (False, False, True): 0.05,
-        (False, False, False): 1.05,
+        (False, True, False): 1.05,
```

Comment:
What is the reason for the changes in this test?

Reply:
This test is changed, because this is not a valid input. String with all entries false are ignored in the construction. Having this is in the testcase was the reason for investigating why it is possible to contract phase poly boxes which do not match the construction from the circuit.

Reply:
I already have some test for that, for example in:
https://github.com/CQCL/tket/blob/feature/add-ppb-from-circuit/pytket/tests/circuit_test.py#L749
https://github.com/CQCL/tket/blob/feature/add-ppb-from-circuit/pytket/tests/circuit_test.py#L762
https://github.com/CQCL/tket/blob/feature/add-ppb-from-circuit/pytket/tests/circuit_test.py#L775

Do you mean something like that? I think there are no other ways possible to construct an invalid box which is not check by this tests.

(Besides that I would suggest removing them from the python tests, I have added a copy of them to c++ to increase the test coverage)

Reply:
Ah yes I forgot about those, thanks.

Reply:
Yes we can remove from the Python if it's tested in the C++.